### DISCUSSION TITLE
Assistance combining collections

### DISCUSSION THREAD

**Discussion by SuperNestendo** [kinkofer/FightClub5eXML#170] [Q&A] [answered]
Almost absolute noob here having hair pulling issues trying to combine these files. I noted the code down the bottom to generate a collection but don't know how to run it or if it will solve my issues.
What I was doing for a good few hours yesterday was combining the xml files in to one big file and importing that. The classes were not working and showing multiple entries in Fight Club so I tried combining them individually and importing that. Still no dice so before I go in and copy paste all the different bit in each xml file I need I was hoping someone could advise on how to run the command at the bottom and if it will let me play a Tashas subclass properly in the app.

**Comment by zcdziura** [comment, marked as answer]
Before you can create your own Compendium file to be imported into Fight Club 5e, you need to install a command-line tool called `xsltproc`. Included in the Utilities folder within this repository is a `merge.xslt` script that will read in entries from a "collection" XML file and then spit out a merged compendium for you to use. Assuming that you're using a Windows computer, you can follow [these instructions](http://www.sagehill.net/docbookxsl/InstallingAProcessor.html) to install the `xsltproc` tool onto your computer.

After you do that, you can follow the instructions for building your own compendium.

**Reply by SuperNestendo** [reply]
Hi @zcdziura 

Thanks for the assistance on this. Unfortunately the link for the instructions seem to have gone down but I was able to find other instructions on [installing xsltproc here](https://pretextbook.org/doc/author-guide/html/installing-xsltproc.html).
Now that I got the program installed, I'm still floored as where to execute the merge. I have tried typing in the line at the bottom of the readme in both Command prompt and powershell from the initial folder where the files are downloaded. I also tried creating a .bat file with this line but no dice as far as I can tell. Sorry to be such a noob and not get this but I am sure I'm missing something very simple.
Also, when running this command, should I have only the one collection I want to create in the collections folder?

Thanks for the assistance again.

**Reply by zcdziura** [reply]
Hi @SuperNestendo 

The instructions in the README are definitely written for folks who "already know" what to do, so I'll write something more detailed here that you can follow. I'll write everything from the perspective that you haven't yet downloaded the repository onto your computer.

Also, any time I reference `Me` in any paths, replace that with your own username. I'm also writing from the perspective of being a Windows user. If you're using a Mac, you'll have to change the path accordingly.

1. Download a ZIP archive of the repository and extract it onto your computer. I have mine extracted to `C:\Users\Me\Documents\FightClub5eXML`
2. Open up a command-line terminal (such as CMD or Powershell) and navigate to the repository: `cd C:\Users\Me\Documents\FightClub5eXML`. This folder is considered the `root` of the repository (like a tree!).
3. Create a new XML file called `Collection.xml`. In this file, you'll create a Collection like the given example found on the README. 
4. Execute the following command: `xsltproc -o BuiltCollection.xml Utilities/merge.xslt Collection.xml`

**Reply by SuperNestendo** [reply]
Hi @zcdziura 

THANK YOU!!!!!! After a few humps and bumps, finally got a complete collection that imported fantastically and can now play a Way of Mercy Monk from my phone without issues.
The issues I ran in to for future reference and anyone else needing help with this in the future:

- Ran the command in last message and wasn't accessing program xsltproc. Had the files sitting on my C:\ drive as per the instruction I linked to but had to copy all those files to my FightClub5eXML folder.
- Spelling.
- I copied one of the collections files from the supplied folder and tried running it as it was after renaming to Collection.xml. No dice. Changed the .. in the file path name to the directory I was working in (eg C:\Users\Me\Downloads\FightClub5eXML-master). That did not work cause I was using \ instead of  /. Changed that over and saved.

After all these little bumps, it ran for a few seconds and produced BuiltCollection.xml in the FightClub5eXML folder. Imported and sucess.
Thanks again for your help and hopefully this helps someone in the future as well.

**Reply by NeonZenith** [reply]
Sorry, beginner at all these here too.

I'm getting stuck with a long list of Warning: failed to load external entity with a list of all the different sources. What am I doing wrong?

**Reply by cjengl** [reply]
Complete noob here how do I download the zip archive of the repository?

**Reply by SuperNestendo** [reply]
On the Main page hit the Code button to bring up a dropdown menu and from there select Download Zip. This will download all the files and their contents.

![image](https://user-images.githubusercontent.com/87750984/139773633-fe5c2f16-5128-4c45-8d17-99e201560e0b.png)

**Reply by Pasatebos** [reply]
all good until i execute the final command. i got this error

![XML](https://user-images.githubusercontent.com/100296576/155343381-79c4e48d-f92c-41d6-80dc-db809f378da8.png)

im doing something wrong but i dont know what.
i creater new .xml named Collection. i copy paste from the collections the text of CoreOnly(thats all i want). but this doesnt work. 1st time for me on all the xml staff... and the README file is not noob friendly!

**Reply by SuperNestendo** [reply]
@Pasatebos 

Had to run this all again cause I deleted all my data off my phone. Was running in to the same issue you are having and figured it out.
In the file path in the xml collection file, "../" means to go up one file from where you currently are and then go to the file path after that. If you remove the "../" from each line (easy done with a find and replace in Notepad++) then this should run and build your collection.
